fuel/datasets/hdf5.py

```diff
@@ -199,5 +213,10 @@ def _out_of_memory_get_data(self, state=None, request=None):
             request = [index + self.subset.start for index in request]
         else:
             raise ValueError
-        return self.filter_sources([data_source[request] for data_source in
-                                    state.values()])
+        rval = []
+        for source_name, data_source in state.items():
+            data = data_source[request]
+            if source_name in self.flatten:
+                data = data.reshape((data.shape[0], -1))
+            rval.append(data)
+        return self.filter_sources(rval)
```

Comment:
A `if source_name not in self.sources: continue` should do the trick as well I guess, and means we don't end up loading and reshaping data unnecessarily, right?